### PR TITLE
Feature/scale bar

### LINKED ISSUE #139: lab view concentration indicator
## Use Case

need it to make it clear what is happening in that view 

## Acceptance Criteria

match the design

### PR DESCRIPTION
**Pull request by meganrm**
Problem
=======
Estimated review size: small

closes #139 
the design has a color scale bar for the lab view 

Solution
========
made a scale bar component that goes from white to yellow. I increased the opacity of the cuvette because the white was looking a little too gray 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. go to molecular view
2. switch back to lab view 
3. press play 

Screenshots (optional):
-----------------------
Screenshot 2024-10-11 at 3 28 17 PM
Screenshot 2024-10-11 at 3 22 12 PM

### CODE REVIEW

File: src/components/ScaleBar.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import styles from "./scalebar.module.css";
+import { MICRO } from "../constants";
+import { SimulariumContext } from "../simulation/context";
+import { AGENT_AB_COLOR } from "../constants/colors";
+
+const ScaleBar: React.FC = () => {
+    const { maxConcentration } = useContext(SimulariumContext);
+    const labelArray = [];
+    for (let i = maxConcentration; i >= 0; i = i - 2) {
```
Comment (toloudis) [inline src/components/ScaleBar.tsx:11]: Is there an assumption that maxConcentration is integer and even?  Do you allow the maxConcentration to change?

**Review by toloudis — APPROVED** [review]

File: src/components/ScaleBar.tsx (hunk 1 above)
Comment (meganrm) [reply src/components/ScaleBar.tsx:11]: it only changes per module. In all likelihood it'll be the same for the other ones, originally I thought I might need very different values for the low affinity and high affinity experiments, but I think they're going to work well in the same range.

File: src/components/ScaleBar.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        labelArray.push(i);
+    }
+    return (
+        <div className={styles.container}>
```
Comment (ShrimpCryptid) [inline src/components/ScaleBar.tsx:15]: Consider adding an `img` role to this div and describing the scale bar it contains with `aria-label`?


e.g.
```suggestion
        <div className={styles.container} role="img" aria-label={`Color key for the cuvette. The lowest concentration (white) is 0 ${MICRO}M and the highest concentration (yellow) is ${maxConcentration} ${MICRO}M.`}>
```

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role

File: src/components/ScaleBar.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                        {i} {MICRO}M -{" "}
+                    </div>
+                ))}
+            </div>
+            <div
+                // inlined this style because the agent color is in javascript
+                // and I want to avoid defining it in both places
+                style={{
+                    background: `linear-gradient(0deg, #ffffff 0%, ${AGENT_AB_COLOR} 100%)`,
+                }}
```
Comment (ShrimpCryptid) [inline src/components/ScaleBar.tsx:28]: Makes sense!

**Review by ShrimpCryptid — APPROVED** [review]
Left a non-blocking accessibility comment but looks sensible to me!

**Pull request merged**